src/handlers/shared/start.ts

```diff
@@ -111,3 +135,19 @@ export async function start(context: Context, issue: Context["payload"]["issue"]
 
   return { output: "Task assigned successfully" };
 }
+
+async function handleTaskLimitChecks(username: string, context: Context, maxConcurrentTasks: number, logger: Context["logger"], sender: string) {
```

Comment:
isn't this code the same as in start function? then you can call this function instead of repeating code